### ISSUE TITLE
Yield event handler, then return event handler misses second event

### ISSUE THREAD

**Issue by masenf** [reflex-dev/reflex#1391]
**Describe the bug**
When yielding events from an event handler, the final returned event is not processed.

**To Reproduce**
```python
import reflex as rx


class State(rx.State):
    def handler(self):
        yield rx.console_log("foo")
        return rx.console_log("bar")



def index() -> rx.Component:
    return rx.fragment(
        rx.vstack(
            rx.button("Log Console", on_click=State.handler),
            spacing="1.5em",
            font_size="2em",
            padding_top="10%",
        ),
    )


app = rx.App(state=State)
app.add_page(index)
app.compile()
```

**Expected behavior**
When the button is clicked, the browser console should display "foo" then "bar".

Instead, we just see "foo".

**Specifics (please complete the following information):**
 - Python Version: 3.11.3
 - Reflex Version: `main`
 - OS: macOS 13.4.1
 - Browser (Optional): Firefox / Chrome

-------------

issue is also observed with non-special events defined on the State as well.

attempting to make the handler that yields into an async generator throws an error:

```console
  File "/Users/masen/code/reflex-dev/repro-event-yield-return/repro_event_yield_return/repro_event_yield_return.py", line 12
    return rx.console_log("bar")
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
SyntaxError: 'return' with value in async generator
```

**Comment by masenf** [comment]
It turns out that the `return` value of a generator is mostly thrown away!

See discussion here: https://stackoverflow.com/questions/34073370/best-way-to-receive-the-return-value-from-a-python-generator

I think this is probably a little known detail of python that will result in confusion as users mix `yield` and `return` haphazardly (like `yield` for state updates, and then ultimately `return` and event handler -- this will NOT work!)

**Comment by masenf** [comment]
Fixed in 0.2.3 via https://github.com/reflex-dev/reflex/pull/1390
